src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/entity/AlertCode.kt

```diff
@@ -48,6 +49,13 @@ data class AlertCode(
       AlertCodeDeactivatedEvent(code, deactivatedAt!!),
     )
   }
+
+  fun reactivate(requestAt: LocalDateTime): AlertCode = this.also {
+    registerEvent(
```

Comment:
It might be cleaner to set `deactivatedAt` and `deactivatedBy` to null in here so that the reactivate business logic is in one place. As the function returns `this`, the calling function can then chain `alertCodeRepository.saveAndFlush(reactivate(alertRequestContext.requestAt)).toAlertCodeModel()`. Up to you though

Reply:
Honestly I haven't looked into why the logic is split between the entity class and the service class.
I'll keep it inline with create/update/deactivate and not change it for the moment. 
If we later decide to refactor this I'll come back to refactor it together with the other operations.